ext/simdjson/src/bindings.cpp

```diff
@@ -0,0 +1,238 @@
+/*
+  +----------------------------------------------------------------------+
+  | simdjson_php                                                         |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 2.0 of the Apache license,    |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | http://www.apache.org/licenses/LICENSE-2.0.html                      |
+  +----------------------------------------------------------------------+
+  | Author: Jinxi Wang  <1054636713@qq.com>                              |
+  | Author: Máté Kocsis <kocsismate@woohoolabs.com>                      |
+  +----------------------------------------------------------------------+
+*/
+
+#include "php.h"
+#include "php_simdjson.h"
+
+#include "simdjson.h"
+#include "bindings.h"
+
+extern "C" {
+	#include <ext/json/php_json.h>
+	#include <Zend/zend_exceptions.h>
+}
+
+// see https://github.com/simdjson/simdjson/blob/master/doc/performance.md#reusing-the-parser-for-maximum-efficiency
+simdjson::dom::parser parser;
```

Comment:
This won't work in thread-safe builds of php, if multiple threads are trying to use the same parser at the same time? https://www.php.net/manual/en/pthreads.requirements.php (I'm not familiar with PHP's tsrm code, but I think it might be possible to do something with thread local storage )

- pthreads was brittle the last time I checked, assuming I'm thinking of the right extension. My concern isn't pthreads, but rather when apache is running multiple php worker threads in a single thread sharing a static variable

Initializing this in the request init (or zeroing it out and manually calling the C++ constructor) and freeing it in request shutdown might be possible